curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da9ea9b847a49fe6ee9b6.md

```diff
@@ -0,0 +1,175 @@
+---
+id: 641da9ea9b847a49fe6ee9b6
+title: Step 25
+challengeType: 0
+dashedName: step-25
+---
+
+# --description--
+
+Now, your FCC Authors Page app is complete. But you need to clean up some things.
+
+After clicking the "Load More Authors" button a couple of times, you'll see it doesn't fetch more authors anymore. That's because you've reached the end of the authors list. For a better UX, that button should be disabled at that point and the text set to something like "No more data to load". You can also set the cursor to `not-allowed` to make sure the user stop clicking it. 
+
+If you look at the authors' bios too, some are longer than the others. you can extract the part of the bios you want and replace the others with an ellipsis (...).
+
+Also, to make things look better you can add a divider line between the authors' images and every other thing in the authors cards.
+
+First, let's start with the "Load More Authors" button. When there's no more data to load, you want to change the button text and disable the button. An `if` statement would be ideal for this.
+
+Inside the `fetchMoreAuthors` function, write an `if` statement and set the condition to `authorDataArr.length <= endingIndex` – meaning there's no more data to load. 
```

Comment:
This seems a bit too long for a single step.

We could break up the explanations of the improvements across a few steps:

```suggestion
Your fCC Authors Page is now complete. But you could improve on a few things.

First, if you click the `Load More Authors` button a couple of time, you'll see that it won't add more authors to the page. That's because you've reached the end of the authors list. For a better user experience, you should make it clear when there's no more data to display by disabling the button and changing its text. An `if` statement is the perfect tool for this.

Inside the `fetchMoreAuthors` function, write an `if` statement and set the condition to `authorDataArr.length <= endingIndex` – meaning there's no more data to load.
```